python/src/aiconfig/schema.py

```diff
@@ -335,6 +335,85 @@ def get_metadata(self, prompt_name: Optional[str] = None):
         else:
             return self.metadata
 
+
+    def get_parameters(
+        self,
+        prompt_or_prompt_name: Optional[str | Prompt] = None,
+    ) -> JSONObject:
+        """
+        Get the parameters for a prompt, using the global parameters if 
+        needed.
+        
+        Args:
+            prompt_or_prompt_name Optional[str | Prompt]: The name of the 
+                prompt or the prompt object. If not specified, use the 
+                global parameters.
+        """
+        prompt = prompt_or_prompt_name
+        if isinstance(prompt_or_prompt_name, str):
+            if prompt_or_prompt_name not in self.prompt_index:
+                raise IndexError(f"Prompt '{prompt_or_prompt_name}' not found in config, available prompts are:\n {list(self.prompt_index.keys())}")
+            prompt = self.prompt_index[prompt_or_prompt_name]
+
+        assert prompt is None or isinstance(prompt, Prompt)
+        if prompt is None or not prompt.metadata or not prompt.metadata.parameters:
+            return self.get_global_parameters()
```

Comment:
It's confusing to return the global parameters when the requested prompt doesn't have any parameters, but otherwise returning only the prompt parameters if they exist. 

In the parameter resolution we actually use a union of global parameters & local parameters (overriding global if they match). Shouldn't this do the same?

Reply:
> It's confusing to return the global parameters when the requested prompt doesn't have any parameters, but otherwise returning only the prompt parameters if they exist.

If the prompt parameters do not exist, it makes sense to check if any are defined in the global params. It's the same concept as https://github.com/lastmile-ai/aiconfig/pull/600

> In the parameter resolution we actually use a union of global parameters & local parameters (overriding global if they match)

Tbh I feel like "check the value closest in scope (prompt) and then check the next closest value in scope (aiconfig)" is more intuitive because it's something a real human would do, but yea it doesn't matter, I can also implement it this way

Reply:
Rename to resolve_prameters or something?

Reply:
Maybe `get_available_parameters` or similar? That avoids the potential for confusion about *which* parameters this refers to; `resolve_parameters` to me means to resolve the parameters that are being used within the prompt input.

And I think this should do the same logic as in `resolve_prompt_string` (less reference/input params) which:
- gets config-level params first
- gets current prompt params, overwriting matching config-level param names with prompt-level value

Otherwise it's pretty confusing to have different behaviours and removes most of the benefits you've listed about reusability / testability.

Reply:
> And I think this should do the same logic as in resolve_prompt_string
> gets config-level params first
> gets current prompt params, overwriting matching config-level param names with prompt-level value

Yea sure, I can do that as well as change the name to `get_available_parameters`! I will fix forward to avoid merge conflicts: https://github.com/lastmile-ai/aiconfig/issues/695